dolphinscheduler-bom/pom.xml

```diff
@@ -35,8 +35,8 @@
         <mybatis-plus.version>3.5.2</mybatis-plus.version>
         <quartz.version>2.3.2</quartz.version>
         <druid.version>1.2.4</druid.version>
-        <zookeeper.version>3.8.0</zookeeper.version>
-        <curator.version>5.3.0</curator.version>
+        <curator.version>4.3.0</curator.version>
+        <zookeeper.version>3.4.14</zookeeper.version>
```

Comment:
1. The doc said that DS support ZooKeeper (3.4.6+). If DS does not support ZooKeeper 3.4 , please update the doc.
2. Currently, ZooKeeper 3.4 has a large number of users.

Reply:
In most cases, we only upgrade the dependencies instead of downgrading them. If you think it is necessary to support ZK version 3.4, you could submit a PR to support both 3.4 and 3.8 instead of downgrading it.

Reply:
This PR support both 3.4 and 3.8

Reply:
@SbloodyS @EricGao888 @danielfree @ke4qqq Any other questions?

Reply:
> In most cases, we only upgrade the dependencies instead of downgrading them. If you think it is necessary to support ZK version 3.4, you could submit a PR to support both 3.4 and 3.8 instead of downgrading it.

+1

Reply:
@SbloodyS  @EricGao888  
1. Usually, upgrading dependencies  versions is easy. But, I did not see the need for this upgrade. 
2. When using ZK  3.4.X, this upgrade caused major problems with not installing and not running. 
3. ZK is part of the infrastructure.Upgrading ZK should be well discussed and should be done carefully
4. So I don't think it should be the "priority:low" label.